scalding-base/src/main/scala/com/twitter/scalding/typed/TypedPipe.scala

```diff
@@ -217,14 +181,9 @@ object TypedPipe extends Serializable {
   final case class MergedTypedPipe[T](left: TypedPipe[T], right: TypedPipe[T]) extends TypedPipe[T]
   final case class ReduceStepPipe[K, V1, V2](@transient reduce: ReduceStep[K, V1, V2])
       extends TypedPipe[(K, V2)]
-  final case class SourcePipe[T](@transient source: TypedSource[T]) extends TypedPipe[T]
+  final case class SourcePipe[T](@transient source: Input[T]) extends TypedPipe[T]
   final case class SumByLocalKeys[K, V](input: TypedPipe[(K, V)], @transient semigroup: Semigroup[V])
       extends TypedPipe[(K, V)]
-  final case class TrappedPipe[T](
```

Comment:
@nownikhil can you check to see if anyone is using `.addTrap` on a TypedPipe?

If they are, you should migrate them off. It is super confusing. If you want to handle errors, use:
```scala
foo.map { input =>
  try Right {
 
  }
  catch {
   case NonFatal(e) =>
     Left { ... }
  }
}
```
which is to say, use Either (which kryo can serialize well) to signal if each row errored or not, and then handle accordingly. I doubt I can clearly and correctly explain how trap really works, and I'm pretty sure that's true of pretty much everyone except Chris Wensel who wrote it in cascading.

Reply:
Pretty sure trapping is discouraged but there might be a few uses. cc @navinvishy @germanschnyder

Reply:
Yeah there are a few uses of this, and the pattern is always the same:
`source.read.addTrap()`